### PR TITLE
add pending authorisation

### PR DESCRIPTION
**Pull request by LBHBomotoso**
## Link to JIRA ticket

https://trello.com/c/GPKJdNkN/13-as-a-contractor-i-can-see-the-status-of-a-variation-so-that-i-know-whether-to-proceed-with-the-work-or-not

## Describe this PR

New Variation: 
Workstatus: on hold
reason: Pending authorisation

### *What is the problem we're trying to solve*

create a new vriation

### *What changes have we introduced*

Update usecase

#### _Checklist_

- [ ] Added end-to-end (i.e. integration) tests where necessary e.g to test complete functionality of a newly added feature
- [ ] Added tests to cover all new production code
- [ ] Added comments to the README or updated relevant documentation _(add link to documentation)_, where necessary.
- [ ] Checked all code for possible refactoring
- [ ] Code pipeline builds correctly

### *Follow up actions after merging PR*

Add approve and reject usecases

### CODE REVIEW

File: RepairsApi/V2/Boundary/Response/WorkOrderListItem.cs [hunk 1]
```diff
@@ -16,6 +16,7 @@ public static class WorkOrderStatus
     public static class WorkOrderReason
```
Comment (kylealexanderhackney) [inline RepairsApi/V2/Boundary/Response/WorkOrderListItem.cs:16]: I appreciate this isnt in this PR but WorkOrderReason doesnt explain much. every new person touching the code will ask "reason for what?" could we name it something that answers that

File: RepairsApi/V2/Infrastructure/Extensions/WorkOrderExtensions.cs [hunk 2]
```diff
@@ -25,7 +25,7 @@ public static string GetAction(this WorkOrder workOrder)
                 //ReasonCode.NoBudget  
                 //ReasonCode.LowPriority 
                 //ReasonCode.FullyFunded 
-                //ReasonCode.PartiallyFunded 
+                ReasonCode.PartiallyFunded => WorkOrderReason.PendAuthorisation,
```
Comment (kylealexanderhackney) [inline RepairsApi/V2/Infrastructure/Extensions/WorkOrderExtensions.cs:28]: if this has already been decided fair enough but PartiallyFunded concerns me as it makes me think payment has been recieved in some form which im assuming isnt actually the case.

File: RepairsApi/V2/UseCase/JobStatusUpdatesUseCases/MoreSpecificSorUseCase.cs [hunk 3]
```diff
@@ -44,7 +44,7 @@ public async Task Execute(JobStatusUpdate jobStatusUpdate)
             if (await _featureManager.IsEnabledAsync(FeatureFlags.SPENDLIMITS) && !authorised.Succeeded)
             {
                 workOrder.StatusCode = WorkStatusCode.Hold;
-                workOrder.Reason = ReasonCode.NoApproval;
+                workOrder.Reason = ReasonCode.PartiallyFunded; //pending authorisation: 40
```
Comment (kylealexanderhackney) [inline RepairsApi/V2/UseCase/JobStatusUpdatesUseCases/MoreSpecificSorUseCase.cs:47]: if we need a comment to explain what this is we should change the  name. ReasonCode.PendingAuthorisation seems more sensible to me but i may be missing something

File: RepairsApi/V2/Boundary/Response/WorkOrderListItem.cs (hunk 1 above)
Comment (LBHBomotoso) [reply RepairsApi/V2/Boundary/Response/WorkOrderListItem.cs:16]: Any suggestions?

File: RepairsApi/V2/Infrastructure/Extensions/WorkOrderExtensions.cs (hunk 2 above)
Comment (LBHBomotoso) [reply RepairsApi/V2/Infrastructure/Extensions/WorkOrderExtensions.cs:28]: Will amend

File: RepairsApi/V2/UseCase/JobStatusUpdatesUseCases/MoreSpecificSorUseCase.cs (hunk 3 above)
Comment (LBHBomotoso) [reply RepairsApi/V2/UseCase/JobStatusUpdatesUseCases/MoreSpecificSorUseCase.cs:47]: will amend

File: RepairsApi/V2/Boundary/Response/WorkOrderListItem.cs (hunk 1 above)
Comment (LBHBomotoso) [reply RepairsApi/V2/Boundary/Response/WorkOrderListItem.cs:16]: I think I will rename it. But I'd like to do the migration for it once the variation workflow is working.

**Review by kylealexanderhackney — APPROVED** [review]

**Pull request closed without merging**